presto-hive/src/main/java/com/facebook/presto/hive/util/ConfigurationUtils.java

```diff
@@ -73,8 +75,17 @@ public static JobConf toJobConf(Configuration conf)
 
     public static JobConf configureCompression(Configuration config, HiveCompressionCodec compression)
     {
-        JobConf result = new JobConf(false);
-        copy(config, result);
+        JobConf result;
+        // Workaround for https://github.com/prestodb/presto-hadoop-apache2/commit/aa3a59101cb14659a96400a3943cd1bb740399b9
```

Comment:
I think we usually don't have commit in code comment? What about the following (feel free to modify it) 

```
FileSystemFactory is used to hack around the abuse of Configuration as a
cache for FileSystem. See FileSystemFactory class for more details.

It is caller's responsibility to create a copy if FileSystemFactory is used.
```

And add javadoc comment to `FileSystemFactory` (perhaps just copy the commit message from https://github.com/prestodb/presto-hadoop-apache2/commit/aa3a59101cb14659a96400a3943cd1bb740399b9)

Reply:
Yeah, i aggree that using commit ids is not what we do. I changed the commit message.

> And add javadoc comment to FileSystemFactory (perhaps just copy the commit message from prestodb/presto-hadoop-apache2@aa3a591)

Do do that we need to create a PR to the `presto-hadoop-apache2`, release the library and update it's version. Do we really think that the comment worth it?